File: client/iih_stub/iih.ci_stub/iih.ci.diag/i/ICidiagMaintainService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using iih.ci.diag.cidiag.d;
using iih.ci.diag.dto.d;
using iih.ci.ord.ems.d;
using iih.en.pv.entdi.d;

namespace iih.ci.diag_stub.i
{
    public interface ICidiagMaintainService
    {

        string saveEntDiag(EntDiDO[] list, CiDiagDO ciDiagDo,string type);
        /// <summary>
        /// 医嘱诊断保存
        /// </summary>
        /// <param name="diDTO"></param>
        /// <returns></returns>
        DIDTO[] SaveCiDiDto(DIDTO[] diDTO,string id,string sd,string tp);
        // 保存诊断
        DIDTO[] SaveCiDiDtos(DIDTO[] diDTO,string des,CiEnContextDTO ciEnContextDto);
        //删除诊断
        CidiagAggDO DeleteCiDiag(CiDiagDO ciDiagDo);
    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/diag/i/ICidiagMaintainService.java
package iih.ci.diag.i;

import iih.ci.diag.cidiag.d.CiDiagDO;
import iih.ci.diag.cidiag.d.CidiagAggDO;
import iih.ci.diag.dto.d.DIDTO;
import iih.ci.ord.ems.d.CiEnContextDTO;
import xap.mw.core.data.BizException;

public  interface ICidiagMaintainService {
	

    /**
     * 医嘱诊断 保存
     * id 诊断过程id
     * sd 诊断过程编码
     * @param diDTO
     * @return
     * @throws BizException
     */
   // public abstract DIDTO[] SaveCiDiDto(DIDTO[] diDTO,String id,String sd,String tp )throws BizException;
    
    public abstract DIDTO[] SaveCiDiDto(DIDTO[] diDTO,String des,CiEnContextDTO cienContextDTO)throws BizException;
    
    
    /**
     * 诊断的保存  （包含事件）
     * @param CidiAggDO
     * @return
     * @throws BizException
     */
    public abstract CidiagAggDO SaveCiDi(CidiagAggDO CidiAggDO)throws BizException;
    /**
     * 删除诊断
     * @param ciDiagDo
     * @return
     * @throws BizException
     */
    public  CidiagAggDO DeleteCiDiag(CiDiagDO ciDiagDo)throws BizException;
}
